x/twap/logic_test.go

```diff
@@ -42,76 +43,287 @@ func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
+	}{
+		"accum with zero value": {
+			record:    newRecord(poolId, time.Unix(1, 0), sdk.NewDec(10), zeroDec, zeroDec),
+			newTime:   time.Unix(2, 0),
+			expRecord: newExpRecord(OneSec.MulInt64(10), OneSec.QuoInt64(10)),
+		},
 		"small starting accumulators": {
-			record:          newRecord(time.Unix(1, 0), sdk.NewDec(10), oneDec, twoDec),
-			interpolateTime: time.Unix(2, 0),
-			expRecord:       newExpRecord(oneDec.Add(OneSec.MulInt64(10)), twoDec.Add(OneSec.QuoInt64(10))),
+			record:    newRecord(poolId, time.Unix(1, 0), sdk.NewDec(10), oneDec, twoDec),
+			newTime:   time.Unix(2, 0),
+			expRecord: newExpRecord(oneDec.Add(OneSec.MulInt64(10)), twoDec.Add(OneSec.QuoInt64(10))),
 		},
 		"larger time interval": {
-			record:          newRecord(time.Unix(11, 0), sdk.NewDec(10), oneDec, twoDec),
-			interpolateTime: time.Unix(55, 0),
-			expRecord:       newExpRecord(oneDec.Add(OneSec.MulInt64(44*10)), twoDec.Add(OneSec.MulInt64(44).QuoInt64(10))),
+			record:    newRecord(poolId, time.Unix(11, 0), sdk.NewDec(10), oneDec, twoDec),
+			newTime:   time.Unix(55, 0),
+			expRecord: newExpRecord(oneDec.Add(OneSec.MulInt64(44*10)), twoDec.Add(OneSec.MulInt64(44).QuoInt64(10))),
 		},
-		"same time": {
-			record:          newRecord(time.Unix(1, 0), sdk.NewDec(10), oneDec, twoDec),
-			interpolateTime: time.Unix(1, 0),
-			expRecord:       newExpRecord(oneDec, twoDec),
+		"same time, accumulator should not change": {
+			record:    newRecord(poolId, time.Unix(1, 0), sdk.NewDec(10), oneDec, twoDec),
+			newTime:   time.Unix(1, 0),
+			expRecord: newExpRecord(oneDec, twoDec),
 		},
 		// TODO: Overflow tests
 	}
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			// correct expected record based off copy/paste values
-			test.expRecord.Time = test.interpolateTime
+			test.expRecord.Time = test.newTime
+			test.expRecord.PoolId = test.record.PoolId
 			test.expRecord.P0LastSpotPrice = test.record.P0LastSpotPrice
 			test.expRecord.P1LastSpotPrice = test.record.P1LastSpotPrice
 
-			gotRecord := twap.RecordWithUpdatedAccumulators(test.record, test.interpolateTime)
+			gotRecord := twap.RecordWithUpdatedAccumulators(test.record, test.newTime)
 			require.Equal(t, test.expRecord, gotRecord)
 		})
 	}
 }
 
-func (s *TestSuite) TestUpdateTwap() {
-	poolId := s.PrepareBalancerPoolWithCoins(defaultTwoAssetCoins...)
-	newSp := sdk.OneDec()
+func (s *TestSuite) TestGetInterpolatedRecord() {
+	baseRecord := newTwapRecordWithDefaults(baseTime, sdk.OneDec(), sdk.OneDec(), sdk.OneDec())
+	// tMin2Record, tMin1Record, baseRecord, tPlus1Record := s.createTestRecordsFromTime(baseTime)
 
 	tests := map[string]struct {
-		record     types.TwapRecord
-		updateTime time.Time
-		expRecord  types.TwapRecord
+		recordsToPreSet     types.TwapRecord
+		testPoolId          uint64
+		testDenom0          string
+		testDenom1          string
+		testTime            time.Time
+		expectedAccumulator sdk.Dec
+		expectedErr         bool
```

Comment:
not blocking, but same here about probably wanting specific errors rather than bool

Reply:
resolved!